lib/src/gitignore.rs

```diff
@@ -198,6 +199,15 @@ impl GitIgnoreFile {
         }
     }
 
+    // TODO: Potentially rename to matchee or matches_path.
```

Comment:
I'm guessing you meant `matches()` (with an 's'). I agree. Could you do that in this commit? Because it's weird to call `matches_file()` on something we know is a directory. Oh, I just found another part of the spec I seem to have missed, namely this:

> If there is a separator at the end of the pattern then the pattern will only match directories, otherwise the pattern can match both files and directories.

So maybe we do want separate `matches_file()` and `matches_dir()` methods, even though they are currently the same. In that case, feel free to leave it as `matches_file()` for now.

We may also want to switch to the [`ignore` crate](https://crates.io/crates/ignore), but that would also be for another PR.

Reply:
> Oh, I just found another part of the spec I seem to have missed, namely this:

Huh, from the code I thought this was accounted for. If you look at working_copy.rs, calls to *matches_file* pass it the result of either *to_internal_file_string* or *to_internal_dir_string*. That way directory paths always end with "/". The regex construction already accounts for that, ie. the lack of "/" at the end adds *$|/.\**.

> We may also want to switch to the [ignore crate](https://crates.io/crates/ignore), but that would also be for another PR.

Was this ignored (no pun intended) previously for some reason, or the existing code just predates it?

Reply:
> Huh, from the code I thought this was accounted for. If you look at working_copy.rs, calls to _matches_file_ pass it the result of either _to_internal_file_string_ or _to_internal_dir_string_. That way directory paths always end with "/". The regex construction already accounts for that, ie. the lack of "/" at the end adds _$|/.*_.
> 

Ah, I see. It's been a while since I wrote this. Thanks for reminding me :) In that case, I would prefer if you can rename the function to just `matches()` in this commit.

> Was this ignored (no pun intended) previously for some reason, or the existing code just predates it?

I don't remember if I tried to find a crate for it before writing this module. It was only recently that @chriskrycho made me aware of it.

Reply:
Renamed to matches. Will take a look at using the ignore crate once this is merged.

Reply:
Thanks. Feel free to merge it

Reply:
Actually, could you add an entry to the changelog before you merge it? This bug is definitely important enough to mention, IMO.